include/envoy/network/proxy_protocol.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "envoy/network/address.h"
+
+namespace Envoy {
+namespace Network {
+
+struct ProxyProtocolData {
+  const Network::Address::InstanceConstSharedPtr& src_addr_;
```

Comment:
This has a high chance to be dangling reference when it is passed around. I'm not sure why our tests didn't catch, do we have integration test for the whole setup?

Reply:
No full setup integration tests at this point. The API (marked as unimplemented) / config factory / TCP Proxy / Doc / etc changes aren't in this PR to try and reduce scope. Those changes + integration would come in a follow up PR. Do you want to change this data type now or in the future PR? This won't be used at all right now.

Reply:
I could also add a TODO to check / fix this for now?

Reply:
At least make this non-reference?

Reply:
Yes I can do that 👍